Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -521,39 +521,88 @@ class DropDownMenu
     var left = details.align === 'left' ? Math.max (0, LayoutUtility.GetOffset (referenceElement).left + details.offsetX) + 'px' : 'auto';
     var right = details.align === 'right' ? Math.max (0, document.documentElement.clientWidth - LayoutUtility.GetOffset (referenceElement).left - referenceElement.offsetWidth - details.offsetX) + 'px' : 'auto';
 
-    const verticalSpacing = parseInt(getComputedStyle(popUpDiv).fontSize) * 2;
+    const popUpMargin = parseInt(getComputedStyle(popUpDiv).fontSize) * 2;
 
     // Save and restore the scrollTop value to retain the scrolling after resizing
     var scrollTop = popUpDiv.scrollTop;
 
     popUpDiv.style.top = top + 'px';
     popUpDiv.style.bottom = 'auto';
-    popUpDiv.style.right = right;
-    popUpDiv.style.left = left;
     popUpDiv.style.position = 'fixed';
 
-    // move dropdown if there is not enough space to fit it on the page
-    if ((LayoutUtility.GetWidth (popUpDiv) > space_left) && (space_left < space_right))
+    // Set to values that allow the popup to expand properly - this way the result of GetWidth is actually correct
+    popUpDiv.style.left = '0';
+    popUpDiv.style.right = 'auto';
+
+    const isContextMenu = details.align === 'left';
+    const requiredPopUpWidth = LayoutUtility.GetWidth(popUpDiv);
+    const dropDownMenuButtonWidth = LayoutUtility.GetWidth(referenceElement);
+    const maxLeftAlignedPopupWidth = (space_right - popUpMargin) + dropDownMenuButtonWidth;
+    const maxRightAlignedPopupWidth = (space_left - popUpMargin) + dropDownMenuButtonWidth;
+    const isPopUpSmallerThanViewPort = requiredPopUpWidth <= (document.documentElement.clientWidth - 2 * popUpMargin);
+
+    const alignToLeftOfReferenceElement = () =>
```

Comment:
@a-ctor I find the solution @iam-mholle has taken query interesting. Given that we now have two complete different approaches for the code used with horizontal and vertical alignment, though, is this requiered due to different technical constraints or should we use the same code style?

My gut feeling says we should align the styles as soon as possible, i.e. in this task or a follow-up task done immediately. Thoughts?

Reply:
Previously, the horizontal positioning was simply a lot simpler than the vertical one. Now both of them are more complex.  The two positioning are separate calculations that do not affect each other.

I don't think it is worth spending the time to align the styles. As they both do their own calculation they do not need to share the same style. If they depended on each other I would also make sure to have a unified code style. That said, we can probably move the two calculations into their own methods. This would make the fact more apparent and improve the readability of the positioning method